File: src/main/scala/ru/tinkoff/gatling/utils/RandomPhoneGenerator.scala
package ru.tinkoff.gatling.utils

import ru.tinkoff.gatling.utils.phone.TypePhone._
import ru.tinkoff.gatling.utils.phone.{Phone, PhoneFormat}

object RandomPhoneGenerator {

  def randomPhone(
      formats: Seq[PhoneFormat],
      typePhone: TypePhone,
  ): String = typePhone match {
    case PhoneNumber         => Phone(formats).phoneNumber
    case TollFreePhoneNumber => Phone(formats).tollFreePhoneNumber
    case E164PhoneNumber     => Phone(formats).e164PhoneNumber
  }

  def randomPhone(
      pathToFormats: String,
      typePhone: TypePhone,
  ): String = typePhone match {
    case PhoneNumber         => Phone(pathToFormats).phoneNumber
    case TollFreePhoneNumber => Phone(pathToFormats).tollFreePhoneNumber
    case E164PhoneNumber     => Phone(pathToFormats).e164PhoneNumber
  }

}

File: src/main/java/ru/tinkoff/gatling/javaapi/utils/RandomPhoneGenerator.java
package ru.tinkoff.gatling.javaapi.utils;

import ru.tinkoff.gatling.utils.phone.*;

import java.util.List;

import static scala.jdk.javaapi.CollectionConverters.asScala;

public final class RandomPhoneGenerator {
    private RandomPhoneGenerator(){}

    public static String randomPhone(List<PhoneFormat> formats, TypePhone.TypePhone typePhone){
        return ru.tinkoff.gatling.utils.RandomPhoneGenerator.randomPhone(asScala(formats).toSeq(), typePhone);
    }

    public static String randomPhone(String pathToFormats, TypePhone.TypePhone typePhone){
        return ru.tinkoff.gatling.utils.RandomPhoneGenerator.randomPhone(pathToFormats, typePhone);
    }
}
